src/elm/interval.coffee

```diff
@@ -157,17 +159,43 @@ module.exports.Collapse = class Collapse extends Expression
     super
 
   exec: (ctx) ->
-    intervals = @execArgs ctx
+    # collapse(argument List<Interval<T>>, per Quantity) List<Interval<T>>
+    [intervals, perWidth] = @execArgs ctx
+
     if intervals?.length <= 1
       intervals
     else
+      # If the per argument is null, the default unit interval for the point type
+      # of the intervals involved will be used (i.e. the interval that has a
+      # width equal to the result of the successor function for the point type).
+      if !perWidth?
+        if intervals[0].low?
+          if intervals[0].low.constructor.name == 'DateTime'
+            # Bonnie will always use milliseconds
+            # TODO determine this dynamically using duration between low and successor
+            perWidth = new Quantity(value: 1, unit: 'millisecond')
+          else if intervals[0].low.isQuantity
+            perWidth = doSubtraction(successor(intervals[0].low), intervals[0].low)
+          else
+            perWidth = successor(intervals[0].low)- intervals[0].low
+        else
+          throw new Error("Point type of intervals provided to collapse cannot be determined.")
+
+        if typeof perWidth is 'number'
+          perWidth = new Quantity(value: perWidth, unit: '1')
+
       # we don't handle imprecise intervals at this time
       for a in intervals
         if a.low.isImprecise?() || a.high.isImprecise?()
           throw new Error("Collapse does not support imprecise dates at this time.")
 
-      # sort intervals by start
-      intervals.sort (a,b)->
+      # Clone intervals so this function remains idempotent
+      intervalsClone = []
+      for interval in intervals
+        intervalsClone.push interval.copy()
```

Comment:
Previously there was a check for `if typeof a.copy == 'function'` before calling `.copy()` is that not needed anymore?

Reply:
`intervals` should only ever be a list of `Intervals`, for which `copy` is defined:
```
  copy: ->
    newLow = @low
    newHigh = @high
    if typeof @low.copy == 'function'
      newLow = @low.copy()
    if typeof @high.copy == 'function'
      newHigh = @high.copy();

    new Interval(newLow, newHigh, @lowClosed, @highClosed)
```